client/layout/guided-tours/config-elements.js

```diff
@@ -116,19 +99,74 @@ export class Step extends Component {
 	}
 
 	componentWillReceiveProps( nextProps, nextContext ) {
+		if ( nextContext.step !== this.context.step ) {
+			this.setSection();
+			debug( 'Step#componentWillReceiveProps: section:', this.section );
+		}
+
 		this.skipIfInvalidContext( nextProps, nextContext );
+		this.quitIfInvalidRoute( nextProps, nextContext );
 		this.scrollContainer.removeEventListener( 'scroll', this.onScrollOrResize );
 		this.scrollContainer = query( nextProps.scrollContainer )[ 0 ] || global.window;
 		this.scrollContainer.addEventListener( 'scroll', this.onScrollOrResize );
 		const shouldScrollTo = nextProps.shouldScrollTo && ( this.props.name !== nextProps.name );
 		this.setStepPosition( nextProps, shouldScrollTo );
 	}
 
+	/* TODO: needed? */
+	shouldComponentUpdate( nextProps, nextState ) {
+		return this.props !== nextProps || this.state !== nextState;
+	}
+
 	componentWillUnmount() {
 		global.window.removeEventListener( 'resize', this.onScrollOrResize );
 		this.scrollContainer.removeEventListener( 'scroll', this.onScrollOrResize );
 	}
 
+	setSection() {
+		// keep track of current section for "blanket exit"
+		// (i.e. quitIfInvalidRoute in `Step` and `Continue`)
+		// TODO(mcsf,lsinger): works only in a few cases and could be more elegant
+		// e.g. doesn't work when switching from reader start to discover
+		// as the Step gets re-created anew
+		// (better than false positives, though)
+		this.section = this.pathToSection( location.pathname );
```

Comment:
Fix this guy here, use proper state instead, probably adding `sectionName` or something to the context.